### DISCUSSION TITLE
Building UIs effortlessly using Ant Design and Kubi

### DISCUSSION THREAD

**Discussion by bytasv** [ant-design/ant-design#44299] [Show and tell]
Hello everyone!

I wanted to share with you an app that I'm building that hopefully could speed up UI development without forcing you to hand-code HTML and CSS and also provide an **easier way to build frontend using Ant Design components**.

The app is very similar to Figma but with the main goal - provide pixel-perfect code for the work that you are doing. It exports react + next.js code so you can seamlessly integrate it with your existing codebase.

Below I recorded a short demo showing some of its capabilities and I invite you to try it yourself. It's currently free for all users so you can give it a go without having to make any hard decisions.

https://github.com/ant-design/ant-design/assets/437214/3928c203-fa2a-46d2-8fb0-90cb782919f8

I'm currently looking for early adopters who would find this app useful and could use it in their day-to-day work or play around building some side projects. I'm offering my own time to help you get onboarded and build your next project with Kubi. The app obviously has bugs here and there, but I'm ready to jump right into the issue and fix it as soon as possible.

Please lemme know if you find this interesting. Maybe you would like to know more? I'm happy to schedule a call or address any questions here in the discussion.

Thank you!

**Comment by kerryj89** [comment]
Wow, that looks amazing!  I landed here on my search for component-based drag and drop builder.  Are we able to import our own design system into this lib (with Custom Elements Manifest maybe it would have an easier time to see the available props)?  So far it seems Grapesjs is the only one that can support web components with some manual work.  I am looking for something like this but they are all coupled with a specific framework, which I understand is for tighter integration from the get-go.  We use StencilJS at my workplace to output to web components and React, Angular, etc. and I know other enterprise level workplaces use Lit also.

But as someone working on the "front of the frontend", it would be nice to just drag and drop components from our design system or for quick mock ups that's 1-to-1 with actual renderings, and hand that off to be wired up.  I think a lot of people would pay for that!

**Reply by bytasv** [reply]
@kerryj89 hey there! Glad you liked it. Actually the whole idea was/is to enable to hook in any design system, basically allowing you to build UI from your own design system without having to write FE code.

The only thing that out-of-box it's not supported yet and we can only do integration manually for now (though that's planned for the future), however if you (or anyone else) would be interested of an integration of their own design system, please let me know :)
